ls1intum/Artemis
# Workaround for null pointer exception in resume exercise on Jenkins setup

### Checklist

### Motivation and Context
Fixes #2600 and #2601 

### Description
Implements a workaround and adds TODOs for future improvements
I also tried to improve the JavaDoc and method parameter names

### Steps for Testing
1. Create and participate in a programming exercise on TS2 (Gitlab / Jenkins) 
2. Cleanup the build plan (not the repository) several times and try out the following methods
3. Resume the exercise (as student) by clicking on resume exercise
4. Resume the exercise by pushing into the repo (the build should automatically get started afterwards and you should still receive a result)
5. Trigger a build as instructor (should automatically resume the exercise)
6. Trigger all builds as instructor (should automatically resume the exercise for multiple users)

**Important:** please review that I did not mix up anything while renaming parameters and variables

### Test Coverage
no changes

### Screenshots
no UI changes
